packages/x-date-pickers/src/internals/hooks/useField/useField.utils.ts

```diff
@@ -820,3 +820,89 @@ export const parseSelectedSections = (
 
   return selectedSections;
 };
+
+export const getSectionValueText = <TDate extends PickerValidDate>(
+  section: FieldSection,
+  utils: MuiPickersAdapter<TDate>,
+  currentDate: TDate | null,
+): string | undefined => {
+  if (!section.value) {
+    return undefined;
+  }
+  switch (section.type) {
+    case 'month': {
+      if (section.contentType === 'digit') {
+        return currentDate && utils.isValid(currentDate)
+          ? utils.format(currentDate, 'month')
+          : utils.format(utils.setMonth(utils.date()!, Number(section.value) - 1), 'month');
+      }
+      const parsedDate = utils.parse(section.value, section.format);
+      return parsedDate ? utils.format(parsedDate, 'month') : undefined;
+    }
+    case 'day':
+      return section.contentType === 'digit'
+        ? utils.format(
+            utils.setDate(utils.startOfYear(utils.date()!), Number(section.value)),
+            'dayOfMonthFull',
+          )
+        : section.value;
+    case 'meridiem':
+      return currentDate && utils.isValid(currentDate)
+        ? utils.format(currentDate, 'meridiem')
+        : undefined;
+    case 'weekDay':
+      return currentDate && utils.isValid(currentDate)
+        ? utils.format(currentDate, 'weekday')
+        : undefined;
+    default:
+      return undefined;
+  }
+};
+
+export const getSectionValueNow = <TDate extends PickerValidDate>(
+  section: FieldSection,
+  utils: MuiPickersAdapter<TDate>,
+  currentDate: TDate | null,
+): number | undefined => {
+  if (!section.value || currentDate === null) {
+    return undefined;
+  }
+  switch (section.type) {
+    case 'weekDay': {
+      if (section.contentType === 'letter') {
+        // TODO: consider handling case when `currentDate` is not valid
+        if (utils.isValid(currentDate)) {
+          return utils.getDayOfWeek(currentDate);
+        }
+        return undefined;
+      }
+      return Number(section.value);
+    }
+    case 'meridiem': {
+      if (!utils.isValid(currentDate)) {
+        const parsedDate = utils.parse(
+          `01:00 ${section.value}`,
+          `${utils.formats.hours12h}:${utils.formats.minutes} ${section.format}`,
```

Comment:
This is a bit interesting method of reliably resolving the time from meridiem.
Do you have better suggestions? 🤔 
Probably another option would be to create a "now" date, set a specific `AM` time, and compare the resulting strings...
However, I'm not sure if it is any better 🤷

Reply:
Would this work?

```tsx
if (formatMeridiem(utils, 'am') === section.value) {
  return 0
} 
if (formatMeridiem(utils, 'pm') === section.value) {
  return 1
}

return undefined
```

Reply:
Maybe, but do you think it is worth creating 2 date objects every time instead of 1 for comparison? 🤔 
The problem is that you can pass `format` to the component without changing the `meridiem` format on the adapter and end up comparing `AM` to `am` in English and what knows what in other languages... 🤷 🙈

Reply:
Right....
Your code is probably safer then :ok_hand: